tests/test_ci/TestControllers.py

```diff
@@ -709,8 +710,235 @@ def test_webhook_pr_opened(self, mock_request, mock_queue_test, mock_github, moc
+        mock_add_test_entry.assert_called_once()
+
+    @mock.patch('mod_ci.controllers.GitHub')
+    @mock.patch('mod_ci.controllers.schedule_test')
+    @mock.patch('requests.get', side_effect=mock_api_request_github)
+    def test_webhook_workflow_run_event_requested_action_with_valid_workflow_name(self, mock_request,
+                                                                                  mock_schedule_test, mock_github):
+        """Test webhook triggered with workflow run event with action requested with a valid workflow name."""
+        data = {'action': 'requested', 'workflow_run': {
+            'name': 'Build CCExtractor on Linux', 'head_sha': 'abcd1234'}}
+        with self.app.test_client() as c:
+            response = c.post(
+                '/start-ci', environ_overrides=WSGI_ENVIRONMENT,
+                data=json.dumps(data), headers=self.generate_header(data, 'workflow_run'))
+        self.assertEqual(response.data, b'{"msg": "EOL"}')
+        mock_schedule_test.assert_called_once()
+
+    @mock.patch('mod_ci.controllers.queue_test')
+    @mock.patch('requests.get', side_effect=mock_api_request_github)
+    def test_webhook_workflow_run_event_completed_action_successful(self, mock_request, mock_queue_test):
+        """Test webhook triggered with workflow run event with action completed and status success."""
+        data = {'action': 'completed',
+                'workflow_run': {'event': 'push',
+                                 'name': 'Build CCExtractor on Linux', 'head_sha': '1',
+                                 'head_branch': 'master'}, 'sender': {'login': 'test_owner'}}
+        fakedata = {'workflow_runs': [
+            {'head_sha': '1', 'status': 'completed',
+                'conclusion': 'success', 'name': Workflow_builds.LINUX},
+            {'head_sha': '1', 'status': 'completed',
+             'conclusion': 'success', 'name': Workflow_builds.WINDOWS}
+        ]}
+
+        class MockedRepository:
+            def statuses(self, *args):
+                return None
+
+            class actions:
+                class runs:
+                    def get(*args, **kwargs):
+                        return fakedata
+
+        class MockedGitHub:
+            def repos(self, *args):
+                return MockedRepository
+
+        with self.app.test_client() as c:
+            from github import GitHub
+            GitHub.repos = Mock(return_value=MockedGitHub.repos)
+
+            response = c.post(
+                '/start-ci', environ_overrides=WSGI_ENVIRONMENT,
+                data=json.dumps(data), headers=self.generate_header(data, 'workflow_run'))
         mock_queue_test.assert_called_once()
 
+    @mock.patch('mod_ci.controllers.deschedule_test')
+    @mock.patch('requests.get', side_effect=mock_api_request_github)
+    def test_webhook_workflow_run_event_completed_action_failure(self, mock_request, mock_deschedule_test):
```

Comment:
```suggestion
    def test_webhook_completed_action_failure(self, mock_request, mock_deschedule_test):
```

Reply:
"workflow_run" being the event name for this event, it shouldn't be skipped I suppose?

Reply:
Sure, please shorten as you feel would be best.